web/api/v1/api_test.go

```diff
@@ -1474,11 +1474,216 @@ func testEndpoints(t *testing.T, api *API, tr *testTargetRetriever, testLabelAPI
 				},
 				errType: errorBadData,
 			},
+			// Start and end before LabelValues starts.
```

Comment:
@brian-brazil  @juliusv 
In all these tests added here, we are only testing series that exist for the entire range via this 
```
load 1m
		test_metric1{foo="bar"} 0+100x100
		test_metric1{foo="boo"} 1+0x100
		test_metric2{foo="boo"} 1+0x100
```

Is it important to test that a record only exists for a portion of the time range? If so, should I add support for that in this PR or do it in another?